third_party/CMakeLists.txt

```diff
@@ -88,3 +88,5 @@ ExternalProject_Add(libsox
   # See https://github.com/pytorch/audio/pull/1026
   CONFIGURE_COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/build_codec_helper.sh ${CMAKE_CURRENT_SOURCE_DIR}/src/libsox/configure ${COMMON_ARGS} --with-lame --with-flac --with-mad --with-oggvorbis --without-alsa --without-coreaudio --without-png --without-oss --without-sndfile --with-opus --with-amrwb --with-amrnb --disable-openmp --without-sndio --without-pulseaudio
 )
+
+add_subdirectory(transducer)
```

Comment:
This one also needs to be guarded. I am trying not to touch RNN transducer with 

```
BUILD_SOX=1 python setup.py develop
```

```
CMake Error at transducer/CMakeLists.txt:32 (ADD_LIBRARY):
  Cannot find source file:

    submodule/src/rnnt_entrypoint.cpp
```

yet it is still trying to build transducer.

The fix is to define an option for CMake like `option(BUILD_TRANSDUCER "Build RNN Transducer" ON)`

Then from `build_tools/setup_helpers/extension.py`, do `cmake -DBUILD_TRANSDUCER=ON|OFF`.

Reply:
Yes, this is the issue I referred to as "Guard building third party transducer even if not added as an extension" in the [follow-ups](https://github.com/pytorch/audio/pull/1137#issuecomment-754410503).

Reply:
Addressed in #1159